### ISSUE TITLE
Pop Shop get freezes all the time.

### ISSUE THREAD

**Issue by rwxe** [pop-os/shop#240]
**Distribution (run `cat /etc/os-release`):**
 NAME="Pop!_OS"
VERSION="20.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.10"
VERSION_ID="20.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=groovy
UBUNTU_CODENAME=groovy
LOGO=distributor-logo-pop-os

**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
pop-shop:
  Installed：3.3.0pop0~1607377061~20.10~996e63e
  Candidate： 3.3.0pop0~1607377061~20.10~996e63e
  Version table：
 *** 3.3.0pop0~1607377061~20.10~996e63e 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu groovy/main amd64 Packages
        100 /var/lib/dpkg/status

**Issue/Bug Description:**
Every time I started the Pop Shop, I can operate normally within the first couple of seconds. After a while, the UI freezes and the window cannot be dragged. After clicking the mouse continuously, it will prompt that the store is not responding, whether to continue to wait or force the end.

**Steps to reproduce (if you know):**
Start the Pop Shop from GUI, and wait for couples of seconds, then it will freeze.

**Expected behavior:**
Should not be freezes.

**Other Notes:**

There is the syslog after started Pop Shop and it freezed.
```
Dec 21 15:01:38 popos systemd[1494]: Started Application launched by gnome-shell.
Dec 21 15:01:38 popos dbus-daemon[740]: [system] Activating via systemd: service name='org.freedesktop.PackageKit' unit='packagekit.service' requested by ':1.210' (uid=1000 pid=12759 comm="io.elementary.appcenter " label="unconfined")
Dec 21 15:01:38 popos systemd[1]: Starting PackageKit Daemon...
Dec 21 15:01:38 popos PackageKit: daemon start
Dec 21 15:01:38 popos dbus-daemon[740]: [system] Successfully activated service 'org.freedesktop.PackageKit'
Dec 21 15:01:38 popos systemd[1]: Started PackageKit Daemon.
Dec 21 15:01:40 popos PackageKit: search-name transaction /1874_dbadecae from uid 1000 finished with success after 313ms
Dec 21 15:01:40 popos PackageKit: get-details transaction /1875_abcccecb from uid 1000 finished with success after 218ms
Dec 21 15:01:41 popos PackageKit: get-packages transaction /1876_bbbdbbca from uid 1000 finished with success after 429ms
Dec 21 15:01:41 popos PackageKit: search-name transaction /1877_bdddcbcc from uid 1000 finished with success after 261ms
Dec 21 15:01:41 popos PackageKit: get-details transaction /1878_bbbbeaeb from uid 1000 finished with success after 235ms
Dec 21 15:01:41 popos PackageKit: search-name transaction /1879_edaddaab from uid 1000 finished with success after 265ms
Dec 21 15:01:42 popos PackageKit: get-details transaction /1880_cbeedbab from uid 1000 finished with success after 222ms
Dec 21 15:01:42 popos PackageKit: search-name transaction /1881_bddaaeed from uid 1000 finished with success after 272ms
Dec 21 15:01:42 popos PackageKit: get-details transaction /1882_ecdcaeaa from uid 1000 finished with success after 222ms
Dec 21 15:01:43 popos PackageKit: search-name transaction /1883_dbedcdde from uid 1000 finished with success after 330ms
Dec 21 15:01:43 popos io.elementary.a[12759]: Package.vala:889: Package not found
Dec 21 15:01:43 popos PackageKit: search-name transaction /1884_bbcabccc from uid 1000 finished with success after 292ms
Dec 21 15:01:43 popos PackageKit: get-details transaction /1885_aecdeaba from uid 1000 finished with success after 230ms
Dec 21 15:01:43 popos PackageKit: search-name transaction /1886_dbaccdbd from uid 1000 finished with success after 275ms
Dec 21 15:01:43 popos io.elementary.a[12759]: Package.vala:889: Package not found
Dec 21 15:01:44 popos PackageKit: search-name transaction /1887_bcceeecb from uid 1000 finished with success after 258ms
Dec 21 15:01:44 popos PackageKit: get-details transaction /1888_ecdcdcdd from uid 1000 finished with success after 323ms
Dec 21 15:01:44 popos PackageKit: search-name transaction /1889_beccceee from uid 1000 finished with success after 278ms
Dec 21 15:01:45 popos PackageKit: get-details transaction /1890_dacaaabe from uid 1000 finished with success after 309ms
Dec 21 15:01:45 popos PackageKit: search-name transaction /1891_eaeeedbb from uid 1000 finished with success after 259ms
Dec 21 15:01:45 popos PackageKit: get-details transaction /1892_cdbceacc from uid 1000 finished with success after 224ms
Dec 21 15:01:45 popos PackageKit: search-name transaction /1893_ebcbdbac from uid 1000 finished with success after 253ms
Dec 21 15:01:46 popos PackageKit: get-details transaction /1894_eabeacde from uid 1000 finished with success after 223ms
Dec 21 15:01:46 popos PackageKit: search-name transaction /1895_cdbdeebb from uid 1000 finished with success after 372ms
Dec 21 15:01:46 popos io.elementary.a[12759]: Package.vala:889: Package not found
Dec 21 15:01:46 popos PackageKit: search-name transaction /1896_bdcbbeab from uid 1000 finished with success after 274ms
Dec 21 15:01:47 popos PackageKit: get-details transaction /1897_bbcddbea from uid 1000 finished with success after 303ms
Dec 21 15:01:47 popos PackageKit: search-name transaction /1898_bedaaccd from uid 1000 finished with success after 357ms
Dec 21 15:01:47 popos PackageKit: get-details transaction /1899_abceaace from uid 1000 finished with success after 224ms
Dec 21 15:01:49 popos PackageKit: refresh-cache transaction /1900_aaedcbcc from uid 1000 finished with success after 1504ms
Dec 21 15:01:56 popos /usr/libexec/gdm-x-session[1622]: (EE) client bug: timer event6 debounce: scheduled expiry is in the past (-8ms), your system is too slow
Dec 21 15:01:56 popos /usr/libexec/gdm-x-session[1622]: (EE) client bug: timer event6 debounce short: scheduled expiry is in the past (-21ms), your system is too slow
Dec 21 15:02:00 popos /usr/libexec/gdm-x-session[1622]: (EE) client bug: timer event6 debounce: scheduled expiry is in the past (-2ms), your system is too slow
Dec 21 15:02:00 popos /usr/libexec/gdm-x-session[1622]: (EE) client bug: timer event6 debounce short: scheduled expiry is in the past (-15ms), your system is too slow
Dec 21 15:02:17 popos systemd[1494]: app-gnome-io.elementary.appcenter-12759.scope: Succeeded.

```
 Any help would be appreciated, thanks.

**Comment by PierreSmile** [comment]
Same issue, and failed to find a solution. Frustrating!

**Comment by vacuum136** [comment]
same issue on my laptop with pop-shop 3.4.2pop0~1617654967~20.04~4589784. But my finding is that the shop works fine if I disconnected the WIFI (only WIFI in my surrounding, so not sure about cable). In this case there is note says "Network Not Available. Connect to..." and there is no pop up window if click "Network Settings" on right side...

**Comment by yonderbread** [comment]
The Pop!_Shop is super laggy for me as well, it also stresses the heck out of my CPU. Most of the stutter seems to happen when switching views and clicking buttons.

**Comment by Tau5** [comment]
Apart from when clicking buttons, the most aggresive freezes is when you open the shop and you start searching without waiting a little

**Comment by Sky4Viper** [comment]
> Apart from when clicking buttons, the most aggresive freezes is when you open the shop and you start searching without waiting a little

same here

**Comment by yonderbread** [comment]
I'm gonna personally go and do a little debugging myself.

**Comment by pktiuk** [comment]
There seems to be a similar issue with upstream: https://github.com/elementary/appcenter/issues/1177.

**Comment by blueberrynotblue** [comment]
has this problem been solved, anyone? T_T

**Comment by zhjh1209** [comment]
run the command in the terminal`io.elementary.appcenter -s`, then open the shop,it will be ok.

**Comment by aha999** [comment]
Same. It freezes usually when i try to search something. Pop!Shop is currently the most frustrating part of Pop!OS imo and i'm using it for about a month...
![image](https://user-images.githubusercontent.com/50620416/130261747-9335d2ab-f581-4cb6-9b8c-47f96db08dd9.png)

**Comment by atrenal** [comment]
I have this same issue, seems like when I open pop shop, it searches for new updates and it freezes the app until it's all loaded.

**Comment by qaqak** [comment]
It happens with me too. I just reopen and it's fine, but the problem happens often. Might as well switch to the Ubuntu Software Center

**Comment by Gadiguibou** [comment]
> It happens with me too. I just reopen and it's fine, but the problem happens often. Might as well switch to the Ubuntu Software Center

That's what I was thinking too, but Pop discourages it as it may break other functionalities...

**Comment by qaqak** [comment]
> but I remember reading that Pop discourages it as it may break other functionalities
What functionalites?

**Comment by Gadiguibou** [comment]
> > but I remember reading that Pop discourages it as it may break other functionalities
> > What functionalites?`

Trying to install `gnome-software` with `apt` yields the following error:

```
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help resolve the situation:

The following packages have unmet dependencies:
 pop-desktop : Depends: sessioninstaller but it is not installable
               Recommends: geary but it is not going to be installed
               Recommends: gedit but it is not going to be installed
               Recommends: gnome-calendar but it is not going to be installed
               Recommends: gnome-contacts but it is not going to be installed
               Recommends: gnome-font-viewer but it is not going to be installed
               Recommends: gnome-power-manager but it is not going to be installed
               Recommends: gnome-weather but it is not going to be installed
               Recommends: io.elementary.sideload but it is not installable
               Recommends: pop-transition but it is not going to be installed
               Recommends: popsicle-gtk but it is not going to be installed
               Recommends: seahorse but it is not going to be installed
               Recommends: totem but it is not going to be installed
               Recommends: file-roller but it is not going to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
```
Here's a Reddit [thread](https://www.reddit.com/r/pop_os/comments/rlzkso/how_do_i_install_gnome_software/) with an answer from a System76 software eng.

**Comment by therealrobster** [comment]
An update on this if it helps anyone.  I have a firewall installed (OpenSnitch) and I saw that pop shop tries to connect to an address called `lh3.googleusercontent.com`.  I blocked it immediately as I don't want my machine talking to any google services.   As soon as I did this, my pop shop exhibits the same issues this thread is describing.  Could it be that you've blocked access to google's servers?  Running a piHole or some other ad blocker stopping network traffic  from hitting google?  Either way, I'm not going to allow it, so the pop shop is bust for me until they remove any talk to google services.   Am thinking of going pure debian to avoid this kind of thing altogether to be honest.

**Comment by mmstick** [comment]
@therealrobster By the way, you can uninstall `pop-shop` and install `gnome-software` with `flatpak-plugin` if you don't like using the shop. That's a less drastic measure than going "pure Debian".

**Comment by geraldmaale** [comment]
I also started exploring Pop OS and the freezing of pop shop drove me crazy. Note that I'm also in China. When I disable my internet, the pop shop works flawlessly but not with the internet enabled.

**My Fix**
I activated my VPN assuming  pop shop wanted to contact Google servers which are prohibited. After activating the VPN and enabling the internet, everything is working as advertised. I could now interact with pop shop to install, update and remove apps.

**Comment by j2l** [comment]
> @therealrobster By the way, you can uninstall `pop-shop` and install `gnome-software` with `gnome-software-plugin-flatpak` if you don't like using the shop. That's a less drastic measure than going "pure Debian".

Thank you @mmstick I didn't know and it's working for me now.

BTW, I 'm pretty sure, people with this issue, like me, are living outside North America.

**Comment by robog-two** [comment]
> > @therealrobster By the way, you can uninstall `pop-shop` and install `gnome-software` with `gnome-software-plugin-flatpak` if you don't like using the shop. That's a less drastic measure than going "pure Debian".
> 
> Thank you @mmstick I didn't know and it's working for me now.
> 
> BTW, I 'm pretty sure, people with this issue, like me, are living outside North America.

Not necessarily! I've had this issue using any VPN (even ones inside NA). I live in the US

**Comment by tavinus** [comment]
Still a problem. Any time it is searching for updates, it becomes unresponsive. So, every time I open it, I know I have to wait until it checks stuff online before it becomes responsive. At least you can wait for the turning wheel to stop and then you know it is ready.  
  
Also, the search field only works at the home screen. Some times it seems to be unresponsive, but you just have to go all the way back to home to be able to use the search field.

**Comment by therealrobster** [comment]
I just use `flatpak update` and the ol `sudo apt update` and `sudo apt upgrade` and just ignore the pop shop

**Comment by MiragianCycle** [comment]
Same issue described here. Not behind a VPN, based in Sri Lanka. The following commands helped: 

`sudo flatpak repair; flatpak repair --user`
`flatpak update`

and regular apt update and upgrades. Basically don't open the popShop. Hope you get this fixed @pop-os

**Comment by User8395** [comment]
I think the problem is the way it handles updates. It seems like the Pop!_Shop runs everything on a single thread (correct me if I'm getting this wrong), which means that it can't run different actions at the same time.

**Comment by Vazno** [comment]
I have the same error

**Comment by User8395** [comment]
Wow, I'm surprised this is still up.

**Comment by tavinus** [comment]
I may be wrong, but it does seem like it gets unresponsive when searching for flatpak updates. I mean, it opens fine, will find some updates (apt) and update the screen with them, then it gets unresponsive for a long time and when it comes back, it will have all the updates, including flatpak ones. I have a lot of flatpaks installed here as well, so it would make sense for it to take a bit more time to find their updates.

In any case, it is running properly on the background and does eventually come back to life with the updates refreshed, but it becomes unresponsive and does not update the interface during that time. This happens 100% of the times here. So, I just usually open it and leave it be while doing something else. After a few minutes I alt TAB to it and update everything. I do this almost every day.

**Comment by robog-two** [comment]
> I may be wrong, but it does seem like it gets unresponsive when searching for flatpak updates. I mean, it opens fine, will find some updates (apt) and update the screen with them, then it gets unresponsive for a long time and when it comes back, it will have all the updates, including flatpak ones. I have a lot of flatpaks installed here as well, so it would make sense for it to take a bit more time to find their updates.  
>   
> In any case, it is running properly on the background and does eventually come back to life with the updates refreshed, but it becomes unresponsive and does not update the interface during that time. This happens 100% of the times here. So, I just usually open it and leave it be while doing something else. After a few minutes I alt TAB to it and update everything. I do this almost every day.  
>   
> EDIT: This is a minor problem, but has already been a reason for me not recommending PopOS to people that are not very good with computers. Some people get freaked with something like this (think it is a virus, or invasion, hardware problem, or whatever). And telling them "this is normal" does not help much...  

Yeah. I think the biggest issue is that the UI freezes and can crash if you continue to interact with it, especially because it's hard to tell if you actually have no updates or if it's just taking an especially long time. I believe there is some rust fetcher thing planned that I think they're waiting on to fix this, though I don't even know if that is still in the works. Also, I'm not sure if this is an issue upstream in Elementary AppCenter? Wondering how/if they've solved this issue and maybe if that's a simpler solution.

https://support.system76.com/articles/roadmap/

**Comment by leviport** [comment]
There is a rebase in the works that should help a lot: https://github.com/pop-os/shop/pull/401
We're working out the last few remaining quirks with it, but it should be ready soon.

**Comment by jacobgkau** [comment]
https://github.com/pop-os/shop/pull/416 should have helped with this at least a bit.

**Comment by pktiuk** [comment]
I think it helped a bit. Others also should test it. 

@rwxe @PierreSmile

**Comment by rwxe** [comment]
@pktiuk Thanks for the reminder, but for some reasons, I still use the old distribution version and cannot easily update the latest software, so I cannot test it, maybe somebody can do that.

**Comment by kevinforrestconnors** [comment]
Same problem.

**Comment by xworks** [comment]
Same problem  +1

**Comment by leviport** [comment]
Pop Shop is being replaced with [Cosmic Store](https://github.com/pop-os/cosmic-store), so this is unlikely to be addressed in the future. 

If you wish to use Cosmic Store now, you can install it with `sudo apt install cosmic-store`. It does not require COSMIC DE in order to run.
